strawberry/extensions/max_aliases.py

```diff
@@ -0,0 +1,74 @@
+from typing import Type
+
+from graphql import (
+    ExecutableDefinitionNode,
+    FieldNode,
+    GraphQLError,
+    InlineFragmentNode,
+    ValidationContext,
+    ValidationRule,
+)
+
+from strawberry.extensions.add_validation_rules import AddValidationRules
+
+
+class MaxAliasesLimiter(AddValidationRules):
+    """
+    Add a validator to limit the number of aliases used.
+
+    Example:
+
+    >>> import strawberry
+    >>> from strawberry.extensions import QueryDepthLimiter
+    >>>
+    >>> schema = strawberry.Schema(
+    ...     Query,
+    ...     extensions=[
+    ...         MaxAliasesLimiter(max_alias_count=15)
+    ...     ]
+    ... )
+
+    Arguments:
+
+    `max_alias_count: int`
+        The maximum number of aliases allowed in a GraphQL document.
+    """
+
+    def __init__(
+        self,
+        max_alias_count: int,
+    ):
+        validator = create_validator(max_alias_count)
+        super().__init__([validator])
+
+
+def create_validator(max_alias_count: int) -> Type[ValidationRule]:
+    class MaxAliasesValidator(ValidationRule):
+        def __init__(self, validation_context: ValidationContext):
+            document = validation_context.document
+            def_that_can_contain_alias = [
+                def_
+                for def_ in document.definitions
+                if isinstance(def_, (ExecutableDefinitionNode))
+            ]
```

Comment:
let's make this a generator 😊

```suggestion
            def_that_can_contain_alias = (
                def_
                for def_ in document.definitions
                if isinstance(def_, (ExecutableDefinitionNode))
            )
```